File: Semestr3/Paradygmaty-Programowania/Cwiczenia/Cwiczenia1/cw5.ml

let rec isPalindrome list = list = List.rev list;;
let p = isPalindrome [2;1;1;2];;



let rec listLength list = 
 if list = [] then 0
 else 1 + listLength (List.tl list);;

let len = listLength([1;2;3]);;
print_int(len)

File: Semestr3/Paradygmaty-Programowania/Cwiczenia/Cwiczenia1/cw5.scala
object cw5 extends App { 

def isPalindrome [A] (list : List[A]) : Boolean = {
  list == list.reverse
}
val isPalindrome: [A] => List[A] => Boolean = [A] => (list: List[A]) => 
    list == list.reverse

def listLength [A] (list: List[A] ) : Int = 
  if list == Nil then 0
  else 1 + listLength(list.tail)


val l = listLength(List(1,2,3))

}
